### ISSUE TITLE
An error occurred while `Style/SymbolArray` cop was inspecting file

### ISSUE THREAD

**Issue by paracycle** [rubocop/rubocop#11026]
After I created an empty gem with `bundle gem` and added some RuboCop rules which include the rule:
```yaml
Style/SymbolArray:
  EnforcedStyle: brackets
```

When I ran `rubocop`, I noticed that the `Style/SymbolArray` cop was failing on the empty symbol array in the generated `Rakefile`, which looked like this:

```ruby
# frozen_string_literal: true

require "bundler/gem_tasks"
task default: %i[]
```

--------

## Expected behavior

I was expecting to see a successful output like the following test case:
```
$ echo "%i[]" | bundle exec rubocop -s test
Inspecting 1 file

Offenses:

test:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
%i[]
^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

## Actual behavior

Instead I am seeing this:
```
$ echo "%i[]" | bundle exec rubocop -s test
Inspecting 1 file
An error occurred while Style/SymbolArray cop was inspecting /Users/ufuk/src/github.com/Shopify/minitest-diff/test:1:0.
To see the complete backtrace run rubocop -d.
C

Offenses:

test:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
%i[]
^

1 file inspected, 1 offense detected, 1 offense autocorrectable

1 error occurred:
An error occurred while Style/SymbolArray cop was inspecting /Users/ufuk/src/github.com/Shopify/minitest-diff/test:1:0.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.1) [arm64-darwin21]
```

<details>
  <summary>Error backtrace with <code>--debug</code></summary>

```
An error occurred while Style/SymbolArray cop was inspecting /Users/ufuk/src/github.com/Shopify/minitest-diff/Rakefile:4:14.
undefined method `loc' for nil:NilClass

        node.source[node.loc.begin.end_pos...node.children[0].loc.expression.begin_pos]
                                                             ^^^^
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/mixin/percent_array.rb:114:in `whitespace_leading'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/mixin/percent_array.rb:86:in `build_bracketed_array_with_appropriate_whitespace'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/style/symbol_array.rb:77:in `build_bracketed_array'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/mixin/percent_array.rb:46:in `check_percent_array'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/style/symbol_array.rb:54:in `on_array'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:100:in `public_send'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:100:in `block (2 levels) in trigger_responding_cops'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:99:in `block in trigger_responding_cops'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:98:in `each'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:98:in `trigger_responding_cops'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:69:in `on_array'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:136:in `block in on_dstr'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:136:in `each'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:136:in `on_dstr'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:71:in `on_pair'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:136:in `block in on_dstr'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:136:in `each'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:136:in `on_dstr'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:71:in `on_hash'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:158:in `block in on_send'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:155:in `each'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:155:in `each_with_index'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:155:in `on_send'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:71:in `on_send'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:136:in `block in on_dstr'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:136:in `each'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:136:in `on_dstr'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/team.rb:155:in `investigate_partial'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cop/team.rb:83:in `investigate'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:315:in `inspect_file'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:286:in `loop'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:255:in `do_inspection_loop'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:138:in `block in file_offenses'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:163:in `file_offense_cache'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:137:in `file_offenses'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:128:in `process_file'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:109:in `block in each_inspected_file'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:108:in `each'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:108:in `reduce'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:108:in `each_inspected_file'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:94:in `inspect_files'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/runner.rb:47:in `run'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cli.rb:72:in `run_command'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cli.rb:79:in `execute_runners'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/lib/rubocop/cli.rb:48:in `run'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/exe/rubocop:19:in `block in <top (required)>'
/opt/rubies/3.1/lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
/Users/ufuk/.gem/ruby/3.1.1/gems/rubocop-1.36.0/exe/rubocop:19:in `<top (required)>'
/Users/ufuk/.gem/ruby/3.1.1/bin/rubocop:25:in `load'
/Users/ufuk/.gem/ruby/3.1.1/bin/rubocop:25:in `<top (required)>'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/cli/exec.rb:58:in `load'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/cli/exec.rb:58:in `kernel_load'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/cli/exec.rb:23:in `run'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/cli.rb:485:in `exec'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/cli.rb:31:in `dispatch'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/cli.rb:25:in `start'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/exe/bundle:48:in `block in <top (required)>'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/lib/bundler/friendly_errors.rb:120:in `with_friendly_errors'
/Users/ufuk/.gem/ruby/3.1.1/gems/bundler-2.3.20/exe/bundle:36:in `<top (required)>'
/Users/ufuk/.gem/ruby/3.1.1/bin/bundle:25:in `load'
/Users/ufuk/.gem/ruby/3.1.1/bin/bundle:25:in `<main>'
```

</details>

## Steps to reproduce the problem

Run `echo "%i[]" | bundle exec rubocop -s test` with the following rule set:
```yaml
Style/SymbolArray:
  EnforcedStyle: brackets
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ bundle exec rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.1) [arm64-darwin21]
```
